src/config/mod.rs

```diff
@@ -17,6 +17,21 @@ static INSTANCE: OnceCell<Config> = OnceCell::new();
 
 const CONFIG_FILE_NAME: &str = "cli.yml";
 
+// This is the default environment name.
+pub const DEFAULT_ENV_NAME: &str = "default";
```

Comment:
You have this value serving two roles. For the most part, we get the default profile name from the server. We could probably eliminate the couple places where we don't. They're all failover cases where we probably could just use `expect` and panic if the value is blank.

That's a minor issue. The real problem is anything in `config` using `"default"` refers to the profile name, not the environment name. They both happen to be "default", but the code is a bit misleading as a result and could break.

Reply:
Chewing this one over a bit more, it may be more idiomatic to make `load_profile` take an `Option<&str>` for the profile name and pass `None` around for the default profile name. Then we can centralize the conversion of the option to the default profile name. I think that would work better with Clap, too, since it returns an `Option<&str>` for the CLI arg value.